### ISSUE TITLE
Exported int var gets broken when set via randi()

### ISSUE THREAD

**Issue by PLyczkowski** [godotengine/godot#10310]
**Operating system or device - Godot version:**

Windows 10 64, Godot 2.1.3

**Issue description:**
I'm setting a new exported variable on a node via:
`export(int) var id`
On creation in editor, I'm setting it to a random value with:
`randomize()`
`node.id = randi()`
Later, at runtime, the id is sometimes different than when viewing in the editor, and generally behaves erratically. Crashes with 'shit? No such file or directory', when the filed is interacted with the value jumps in editor etc.

**Comment by Zylann** [comment]
Where are you doing this `node.id = randi()`?
If you do that in `_ready()`, it will obviously set to a different value everytime, no matter what you do in editor.
You should also check that your assignment isn't set a second time in editor due to another call to `_ready()` for example.

**Comment by PLyczkowski** [comment]
The randi() is set once, though it seems it sets the exported var to a value bigger that it can handle, and then it starts misbehaving:

![2017-08-12_22-26-06](https://user-images.githubusercontent.com/7307968/29243998-b1a18a8e-7fad-11e7-9d33-08b50326af52.gif)

**Comment by Zylann** [comment]
Note that randi() returns a value between 0 and 2^32. But due to GDScript values being variants, it will be stored as a signed integer, which handles values between -2^31 and 2^31 (Values above 2^32 will be negative).

**Comment by PLyczkowski** [comment]
Seems also that breaking a single export var can break other export vars in file or object. The scene I'm working on crashed with 'shit? No such file or directory, Server disconnected', then I manually set other export int vars to the same value they displayed, and the scene started working again..

**Comment by PLyczkowski** [comment]
I have a custom node type added via an addon (A). Node A on _ready in editor (after being added to the scene) adds a new node next to it (B), that has an exported var called id. Node A then set's the id of node B to randi(). Then node A removed itself (so the id is set only once).

My assumption is that it sometimes sets id too high, and that somehow makes the exported var broken. It also breaks other, unrelated export int vars.

**Comment by PLyczkowski** [comment]
And the fact that export vars, especially ones unrelated to the problem, can get corrupted that easily, is very disconcerting - one can store important data in them, like the totality of the in-game dialogues for example..

**Comment by PLyczkowski** [comment]
Or I'm doing something wrong. Exceeding the export var value like on the above gif feels a bit buggy though.

**Comment by kubecz3k** [comment]
I was experimenting with this for a moment but was not able to reproduce the crash problem. @PLyczkowski I think you really should create minimal reproduction project.

**Comment by kubecz3k** [comment]
Or provide a detailed list of steps that are needed to reproduce.

**Comment by kubecz3k** [comment]
Also in the process you might find out more directly where the problem lies, feel encouraged to close this issue and start freshly with more direct one.

**Comment by PLyczkowski** [comment]
Alright, I'll try to pinpoint the problem more precisely, and start a new issue when I do.
